File: src/main/java/app/askresume/api/resume/facade/ResumeFacade.java
package app.askresume.api.resume.facade;

import app.askresume.api.resume.dto.request.GenerateExpectedQuestionRequest;
import app.askresume.api.resume.dto.request.ResumeDataRequest;
import app.askresume.api.resume.dto.response.WhatGeneratedResponse;
import app.askresume.api.resume.mapper.CareerYearMapper;
import app.askresume.api.resume.mapper.GenerateExpectedQuestionMapper;
import app.askresume.domain.prompt.constant.PromptType;
import app.askresume.domain.prompt.service.PromptService;
import app.askresume.external.gpt.service.GptService;
import app.askresume.domain.job.service.JobService;
import app.askresume.domain.resume.service.ResumeService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

import java.util.List;

@Slf4j
@Service
@RequiredArgsConstructor
public class ResumeFacade {

    private final ResumeService resumeService;
    private final GptService gptService;
    private final JobService jobService;
    private final GenerateExpectedQuestionMapper generateExpectedQuestionMapper;
    private final CareerYearMapper careerYearMapper;

    private final PromptService promptService;


    public WhatGeneratedResponse generate(final GenerateExpectedQuestionRequest request) {

        log.debug("이력서 생성 update request : {}", request.toString());

        final String job = jobService.findJobNameById(request.jobId()).getMasterName();
        final String difficulty = request.difficulty();
        final String careerYear = careerYearMapper.toCareer(request.careerYear());
        final String locale = request.locale();
        final List<ResumeDataRequest> resumeData = generateExpectedQuestionMapper.toResumeData(request.contents());

        String prompt = promptService.findByPromptType(PromptType.QUESTIONS_AND_MODEL_ANSWERS);

        WhatGeneratedResponse response = gptService.createdExpectedQuestionsAndAnswer(prompt, job, difficulty, careerYear, locale, resumeData);


        // request 내용, 생성된 질문 저장

        // 리스폰
        return response;
    }

}

File: src/main/kotlin/app/askresume/api/resume/facade/ResumeFacade.kt
package app.askresume.api.resume.facade

import app.askresume.api.resume.dto.request.GenerateExpectedQuestionRequest
import app.askresume.api.resume.dto.response.WhatGeneratedResponse
import app.askresume.api.resume.mapper.CareerYearMapper
import app.askresume.api.resume.mapper.GenerateExpectedQuestionMapper
import app.askresume.domain.job.service.JobService
import app.askresume.domain.prompt.constant.PromptType
import app.askresume.domain.prompt.service.PromptService
import app.askresume.domain.resume.service.ResumeService
import app.askresume.external.gpt.service.GptService
import app.askresume.global.util.LoggerUtil.log
import org.springframework.stereotype.Service


@Service
class ResumeFacade(
    private val resumeService: ResumeService,
    private val gptService: GptService,
    private val jobService: JobService,
    private val generateExpectedQuestionMapper: GenerateExpectedQuestionMapper,
    private val careerYearMapper: CareerYearMapper,
    private val promptService: PromptService,
) {

    val log = log()

    fun generate(request: GenerateExpectedQuestionRequest): WhatGeneratedResponse {
        val job = jobService.findJobNameById(request.jobId).masterName
        val difficulty = request.difficulty
        val careerYear = careerYearMapper.toCareer(request.careerYear)
        val locale = request.locale
        val resumeData = generateExpectedQuestionMapper.toResumeData(request.contents)

        val prompt = promptService.findByPromptType(PromptType.QUESTIONS_AND_MODEL_ANSWERS)

        // request 내용, 생성된 질문 저장

        // 리스폰
        return gptService.createdExpectedQuestionsAndAnswer(prompt, job, difficulty, careerYear, locale, resumeData)
    }
}